src/Core/Grid/Column/Type/CartTotalColumn.php

```diff
@@ -0,0 +1,59 @@
+<?php
+/**
+ * 2007-2019 PrestaShop and Contributors
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2019 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\PrestaShop\Core\Grid\Column\Type;
+
+use PrestaShop\PrestaShop\Core\Grid\Column\AbstractColumn;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+
+/**
+ * Defines column that displays cart total price
+ */
+final class CartTotalColumn extends AbstractColumn
```

Comment:
Maybe we can make it more generic so it can be reused for other data rows that have a total, like orders ? Something like a `TotalColumn` ?

Reply:
Well, this is a bit specific, as it has `is_order_placed_field` option.

Reply:
Yes but it is used as a condition to trigger a specific markup to be used
```
{% if record[column.options.is_order_placed_field]%}
    <span class="badge badge-success rounded">
      {{ record[column.options.field] }}
    </span>
  {% else %}
    {{ record[column.options.field] }}
  {% endif %}
```
can become
```
{% if record[column.options.field_must_be_highlighted]%}
    <span class="badge {{ record[column.options.field_badge] }} rounded">
      {{ record[column.options.field] }}
    </span>
  {% else %}
    {{ record[column.options.field] }}
  {% endif %}
```
if we use a generic option name like `field_must_be_highlighted` an an additionnal `field_badge` which must be one of `badge-success`, `badge-warning`, ...

Reply:
That would work for me, but...

I think I can recall having discussion about it with @mickaelandrieu, he didn't want us to have columns that are "view specific", like `ColorColumn`, `HighlightedColumn` or anything that describes how it looks. Same with column properties. 

I'll implement whichever you want. :)

Reply:
Mmmm 🤔
That's a tough one. Mickael's point of view is that Columns should remain an abstract data model and let Twig handle the rendering.

Here's my suggestion: we provide a `field_must_be_highlighted` boolean option that you can set to `true`, and when set to `true` it uses a 2nd option `highlight_level`.

`highlight_level` is a string option, default is "default", and it can accept values such as "success", "warning", ...
So the php class Column does not control how it is rendered. But on twig side we match the option with badges. And a developer who wishes to implement it another way can override the Twig template, for example by adding a border, or a background color, or a tooltip.

So the column carries the information "this data needs to be highlighted somehow, at this level" and the Twig template chooses how it implements this particular request

Wdyt ?

Reply:
Yeah, but that does not sound like a `CartTotalColumn` nor `TotalColumn` anymore, should it be named `HighlightedColumn`? If so, it already describes the "view" part. :/

Reply:
`TotalColumn` refers to the meaning of the data being presented in the column, not how it is displayed.

The more I discuss this, the more I think we are overthinking it. We have LinkColumn, DraggableColumn, BadgeColumn. It's already very related to the view. Let's assume and go for HighlightedColumn.

Just keep quiet so mickaelandrieu will not see it :trollface: